System.IO.Abstractions.TestingHelpers/MockFile.cs

```diff
@@ -57,8 +55,9 @@ public override void AppendAllText(string path, string contents, Encoding encodi
 
             if (!mockFileDataAccessor.FileExists(path))
             {
-                var mockPath = mockFileDataAccessor.Path;
-                var dir = mockPath.GetDirectoryName(mockPath.GetFullPath(path));
+                var dir = mockFileDataAccessor.Path.GetDirectoryName(
```

Comment:
I prefer the current version, where we keep a temporary `mockPath` to reduce duplications

Reply:
I had made a similar change in 2 other places in that same file. Looking at it again, I noticed in all 3 places, it's doing the same thing: get the directory name of the full path and raise a DirectoryNotFoundException if that directory doesn't exist. Added a commit bcc061cd550d0341b4c6ffafdecb48c75946c25d containing the de-triplication of those 3 checks.

I now see a pre-existing `VerifyDirectoryExists` used in 3 other places, it appears it might do the same thing. If it does, I'll have all 6 call sites use just 1 method and push that.

Reply:
Cool, sure!

Reply:
They behave the same, so I combined them.